src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs

```diff
@@ -156,6 +162,26 @@ private static bool IsLocalUsedAsArgumentForJsonSerializerOnly(IObjectCreationOp
             return operation;
         }
 
+        private static bool IsLocalReferenceInsideChildLoop(ILocalReferenceOperation localRef, IBlockOperation symbolBlock)
+        {
+            IOperation? current = localRef;
+            while ((current = current?.Parent) is not null)
+            {
+                if (current is ILoopOperation loop)
```

Comment:
I'm guessing this answers my earlier question in https://github.com/dotnet/roslyn-analyzers/pull/6979/files#r1358073471

Reply:
I'll try to improve the comment.